USFMToolsSharp/HtmlRenderer.cs

```diff
@@ -26,19 +48,52 @@ public string Render(USFMDocument input)
                 output.Append($"<meta charset=\"{encoding}\">");
             }
             output.AppendLine("<link rel=\"stylesheet\" href=\"style.css\">");
-            output.AppendLine("</head>");
-            output.AppendLine("<body>");
 
-            foreach(Marker marker in input.Contents)
+            if (InsertedHTML.Contains("</head>"))
+            {
+                output.AppendLine(InsertedHTML);
+            }
+            else {
+                output.AppendLine("</head>");
+            }
+            
+            // HTML tags can only have one class, when render to docx
+            output.AppendLine($"<body class=\"{(isSingleSpaced ? "" : "double-space")}\">");
+            output.AppendLine($"<div class=\"{ (hasOneColumn ? "" : "multi-column")}\">");
+            output.AppendLine($"<div class=\"{ (isTextJustified ? "justified" : "")}\"> ");
+            output.AppendLine($"<div class=\"{ (isL2RDirection ? "" : "rtl-direct")}\"> ");
+
+            
+
+            foreach (Marker marker in input.Contents)
             {
                 output.Append(RenderMarker(marker));
             }
 
+            output.AppendLine(InsertedFooter);
+            output.AppendLine(RenderFootnotes());
+
+            output.AppendLine("</div>");
+            output.AppendLine("</div>");
+            output.AppendLine("</div>");
+            output.AppendLine("</div>");
             output.AppendLine("</body>");
             output.AppendLine("</html>");
             return output.ToString();
         }
 
+        public void InsertFooters(string input)

```

Comment:
I wouldn't do this. I would either set this as a public field or as a property with a getter and setter. That is typically the best way to go.

Reply:
And it should be Footer not Footers although that's pretty minor.